int64 num_node
